src/webview/components/memory-table.tsx

```diff
@@ -220,33 +270,76 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
             tableStyle: { minWidth: '30rem' },
             value: rows
         };
+    }
+
+    protected createScrollingBehaviorSpecificProperties(props: DataTableProps<MemoryRowData[]>): DataTableProps<MemoryRowData[]> {
+        if (this.props.scrollingBehavior === 'Auto-Append') {
+            return {
+                ...props,
+                pt: {
+                    wrapper: {
+                        onScroll: event => {
+                            if (!this.isLoading) {
+                                // Append new data only if we reach the bottom
+                                const element = event.target as HTMLDivElement;
+                                const distanceBottom = Math.abs(element.scrollHeight - element.scrollTop - element.clientHeight);
+
+                                if (distanceBottom < 1) {
+                                    this.appendMoreMemory();
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        } else {
+            return {
+                ...props,
+                footer: this.renderFooter()
+            };
+        }
+    }
+
+    /**
+     * This method asserts that we have sufficient rows visible to enable vertical scrollbars
+     */
+    protected assertSufficientVisibleRowsForScrollbar(): void {
```

Comment:
As this method has side-effects and doesn't return a `boolean`, I find the name starting with `assertXX` a bit surprising. Maybe `ensureXX`?